### PR TITLE
adding accessorFactory in PrecompilePricingUtils

### PR DESCRIPTION
**Pull request by tdermendzhievv**
**Description**:
This PR enables obtaining of the right txn accessor in PrecompilePricingUtils class. This change is needed for correct calculation of the precompile transaction fees. This change is continuation of #3716  

**Related issue(s)**:

Fixes #

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Comment by stoqnkpL** [comment]
We should never have an invalid Transaction object in the precompiles resulting in an `InvalidProtocolBufferException` exception being thrown by [constructSpecializedAccessor](https://github.com/hashgraph/hedera-services/blob/0f3475d1fbeb16b6f7a2fcec5585eb3df1f8ba97/hedera-node/src/main/java/com/hedera/services/utils/accessors/AccessorFactory.java#L74) when we're calling [gasFeeInTinybars](https://github.com/hashgraph/hedera-services/blob/76f162e654dfb708ce355a63571a06dea6c7551e/hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/utils/PrecompilePricingUtils.java#L135). I think we can simplify this PR by:
- not throwing `InvalidProtocolBufferException` in `PrecompilePricingUtils.gasFeeInTinybars()`
- instead if `accessorFactory.constructSpecializedAccessor(txn.toByteArray())` throws an exception, we catch it and throw an `InvalidTransactionException` right away

@tinker-michaelj wdyt

File: hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/utils/PrecompilePricingUtils.java [hunk 1]
```diff
@@ -142,7 +147,7 @@ public long gasFeeInTinybars(
                         .setSignedTransactionBytes(signedTxn.toByteString())
                         .build();
 
-        final var accessor = SignedTxnAccessor.uncheckedFrom(txn);
+        final var accessor = accessorFactory.constructSpecializedAccessor(txn.toByteArray());
```
Comment (Neeharika-Sompalli) [inline hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/utils/PrecompilePricingUtils.java:150]: May be we could add `uncheckedSpecializedAccessor` that catches `Exception` in `AccessorFactory` and throws `IllegalArgumentException` similar to `SignedTxnAccessor.uncheckedForm`

```
    public SignedTxnAccessor uncheckedSpecializedAccessor(byte[] signedTxnWrapperBytes) {
        try {
            return constructSpecializedAccessor(signedTxnWrapperBytes);
        } catch (InvalidProtocolBufferException e) {
            throw new IllegalArgumentException("Not a valid signed transaction");
        }
    }

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM !

**Review by Nana-EC — APPROVED** [review]
LG

**Comment by shemnon** [comment]
Can you give a concrete example for why this  ultimate call to `SignedTxnAccessor.from` needs to be wrapped up in a dagger injectable instance?

**Comment by tdermendzhievv** [comment]
> Can you give a concrete example for why this ultimate call to `SignedTxnAccessor.from` needs to be wrapped up in a dagger injectable instance?

As I have described in the description of this PR my change is an adjustment to the functionalities that were introduced in the [linked issue](https://github.com/hashgraph/hedera-services/pull/3716).

**Comment by Neeharika-Sompalli** [comment]
> As I have described in the description of this PR my change is an adjustment to the functionalities that were introduced in the [linked issue](https://github.com/hashgraph/hedera-services/pull/3716).

@tdermendzhievv  @shemnon  [this linked PR](https://github.com/hashgraph/hedera-services/pull/3716),  was a step taken to support aliases in all queries and transactions (which is postponed to future release), by creating custom accessors for each transaction type.

**Review by shemnon — APPROVED** [review]
@tdermendzhievv's response was basically "see the linked pr" - which I had looked at before and it wasn't clear.  I was asking for a concrete example not a direction to re-read the other PR.  @Neeharika-Sompalli provided that concrete example.

**Pull request merged**